src/peer-store/persistent/index.js

```diff
@@ -58,6 +65,7 @@ class PersistentPeerStore extends PeerStore {
     this.on('change:protocols', this._addDirtyPeer)
     this.on('change:multiaddrs', this._addDirtyPeer)
     this.on('change:pubkey', this._addDirtyPeer)
+    this.on('change:metadata', this._addDirtyPeerMetadata)
```

Comment:
Why don't we do the same thing as the other events and just mark the peer itself as dirty, then get the latest when we go to commit?

Reply:
There is one issue with that approach, that I cannot circumvent unless I add the extra dirty record. 

While in the other books we know that if something changed and it is not in the book anymore, it is deleted, in the metadataBook is not that simple. So, for adding it is simple, but for deleting we would not know the metadata keys that were deleted (because they were simply gone from the book), which would result in metadata being persisted and not removed anymore. When we do the restart, that metadata would be around.

Reply:
Fair enough, we could check the datastore keys and then delete the ones that we no longer have, but this might have some drawbacks to shutdown storage time (extra datastore round trips). We can punt this decision for now, as there's no user facing impact with this.